### PR TITLE
7 code refactor

### LINKED ISSUE #7: Code refactor
A couple of code refactoring ideas from @ddriddle, proposed in #4 :

image

image

### PR DESCRIPTION
**Pull request by zdc217**
- Add a basic decorator to handle exceptions
- Refactor some unneeded logic out of get_clearpass_token
- Add separate exception handling for canary
    - We want it to handle exceptions a little differently than other endpoints

Closes #7

### CODE REVIEW

File: app.py
Comment (edthedev) [inline app.py]: I like to call this the @#$%! TODO. Rather not merge this with a silent suppress here. Can we at log a message or not catch?

**Review by edthedev — CHANGES_REQUESTED** [review]
I'm not even sure that wrapper is in use, but it still makes me nervous. Can we make it re-throw, or not catch, or log before we merge?

File: app.py
Comment (zdc217) [reply app.py]: Logging to stderr now. The default behavior in a docker container is for stderr to write to a json logfile on the local host system. While the container is running the log can be accessed with the ```$ docker logs <CONTAINER ID>``` command. This can be changed to a different driver when going to prod or when the code is on a separate test system so that we can persist/view logs.

**Review by edthedev — APPROVED** [review]

**Review by hfsilberg — APPROVED** [review]

**Review by mpitcel — APPROVED** [review]

**Pull request merged**
